src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.Fixer.cs

```diff
@@ -21,9 +29,51 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
+            var diagnostic = context.Diagnostics[0];
+            var codeAction = CodeAction.Create("Use indexer", c => UseCollectionDirectly(context.Document, context.Span, c));
+            context.RegisterCodeFix(codeAction, diagnostic);
+
             // Fixer not yet implemented.
             return Task.CompletedTask;
 
         }
+
+        private async Task<Document> UseCollectionDirectly(Document document, TextSpan span, CancellationToken c)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(c).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(c).ConfigureAwait(false);
+
+            var invocationSyntax = GetSyntaxOfType<InvocationExpressionSyntax>(syntaxRoot.FindNode(span));
+            var invocationOp = (IInvocationOperation)semanticModel.GetOperation(invocationSyntax);
+
+            //implemented for "Enumerable.First" only
+            if (invocationOp.TargetMethod.Name != "First")
+            {
+                return document;
+            }
+
+            var collectionSyntax = GetSyntaxOfType<ExpressionSyntax>(invocationOp.Arguments[0].Syntax);
```

Comment:
Method name being First does not guarantee that it has non-zero arguments. This code needs stronger guard.

Reply:
@mavasani this particular check is already done in the analyzer itself:

https://github.com/dotnet/roslyn-analyzers/blob/a44670f717ad86b71f2aeb2aa910c4d726732874/src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.cs#L138

Does it still need to be checked in the fixer too?

P.S. Thanks for all these reviews. I'll fix them soon.

Reply:
Yes, fixers should not assume anything about the analyzer and/or the diagnostic location or properties. For example, you could have a completely different analyzer package reporting the same diagnostic ID, and these diagnostics will also get handed to the fixer.

Reply:
Addressed in the comment below.

Reply:
Add a check that you have at least one argument.